counterparty-core/counterpartycore/lib/messages/utxo.py

```diff
@@ -160,9 +243,14 @@ def parse(db, tx, message):
 
     # detach if source is a UTXO
     if util.is_utxo_format(source):
-        source_address, _value = backend.bitcoind.get_utxo_address_and_value(source)
-        if source_address != tx["source"]:
-            problems.append("source does not match the UTXO source")
+        if util.enabled("spend_utxo_to_detach"):
```

Comment:
Can you comment this code a bit? It's still pretty opaque to me.

Reply:
done!